src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1461,6 +1507,65 @@ private RefreshItem getRefreshToken(final AuthenticationRequest request) {
         return refreshItem;
     }
 
+    private TokenCacheItem findFamilyItemForUser(final UserIdentifierType userIdentifierType, final String userId)
+    {
+        final String methodName = ":findFamilyItemForUser";
+            
+        if (StringExtensions.IsNullOrBlank(userId))
+        {
+            Logger.v(TAG + methodName, "User id is not provided, cannot continue with finding family item.");
+            return null;
+        }
+
+        Iterator<TokenCacheItem> allItems = mTokenCacheStore.getAll();
+        if (allItems == null || !allItems.hasNext())
+        {
+            Logger.v(TAG + methodName, "No items in the cache, cannot continue with finding family item.");
+        }
+        
+        if (userIdentifierType == UserIdentifierType.UniqueId)
+        {
+            Logger.v(TAG + methodName, "UserIdentifier type is unique id, will look for the family item based on unique id.");
+            while (allItems.hasNext())
+            {
+                final TokenCacheItem tokenCacheItem = allItems.next();
+                final UserInfo tokenUserInfo = tokenCacheItem.getUserInfo();
+                if (tokenUserInfo!= null 
+                    && tokenUserInfo.getUserId() != null 
+                    && tokenUserInfo.getUserId().equalsIgnoreCase(userId))
+                {
+                    if (!StringExtensions.IsNullOrBlank(tokenCacheItem.getFamilyClientId())
```

Comment:
family id today is only a flag(it's just 1 today) returned from Evo sts, it's across tenant, across authority.  We also don't return it back to client. When we receive a token response from server with FoCi marked, we'll store it in the cache. For client using us, you acquire token following the way you were following today. If we don't find regular refresh token or mrrt, we will try to find a token cache with FoCI marked, and if found, we will try to use it.